addons/address_autocomplete/models/res_partner.py

```diff
@@ -0,0 +1,16 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def _get_view(self, view_id=None, view_type='form', **options):
+        arch, view = super()._get_view(view_id, view_type, **options)
+
+        if view_type == 'form':
+            for node in arch.xpath("//field[@name='street']"):
+                node.set('widget', 'address_autocomplete')
```

Comment:
Not a fan of doing this in py instead of xml. At the very least, I would only apply the widget if no widget is set on the node (i.e. fallback, instead of forcing that widget)